pkg/scheduler/framework/plugins/interpodaffinity/plugin_test.go

```diff
@@ -0,0 +1,203 @@
+			newPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name: "update a pod from non-match to match pod's affinity - multiple terms case",
+			pod: st.MakePod().Name("p").PodAffinityExists("aaa", "hostname", st.PodAffinityWithRequiredReq).
+				PodAffinityExists("bbb", "hostname", st.PodAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("aaa", "").Label("bbb", "").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.Queue,
+		},
+		{
+			name:         "modify pod label to change it from satisfying pod anti-affinity to not satisfying anti-affinity",
+			pod:          st.MakePod().Name("p").PodAntiAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAntiAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("service", "aaaa").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			expectedHint: framework.Queue,
+		},
+		{
+			name:         "modify pod label to change it from not satisfying pod anti-affinity to satisfying anti-affinity",
+			pod:          st.MakePod().Name("p").PodAntiAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAntiAffinityWithRequiredReq).Obj(),
+			newPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("service", "bbb").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name:         "delete a pod which doesn't match pod's anti-affinity",
+			pod:          st.MakePod().Name("p").PodAntiAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAntiAffinityWithRequiredReq).Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("service", "securityscan").Obj(),
+			expectedHint: framework.QueueSkip,
+		},
+		{
+			name:         "delete a pod which matches pod's anti-affinity",
+			pod:          st.MakePod().Name("p").PodAntiAffinityIn("service", "region", []string{"securityscan", "value2"}, st.PodAntiAffinityWithRequiredReq).Obj(),
+			oldPod:       st.MakePod().Node("fake-node").Label("aaa", "a").Obj(),
+			expectedHint: framework.Queue,
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			logger, ctx := ktesting.NewTestContext(t)
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			snapshot := cache.NewSnapshot(nil, nil)
+			pl := plugintesting.SetupPluginWithInformers(ctx, t, New, &config.InterPodAffinityArgs{}, snapshot, namespaces)
+			p := pl.(*InterPodAffinity)
+			actualHint, err := p.isSchedulableAfterPodChange(logger, tc.pod, tc.oldPod, tc.newPod)
+			if tc.expectedErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedHint, actualHint)
```

Comment:
Can we _not_ use this kind of testing library and use `cmp.Diff` instead, in the area of SIG/Scheduling?
In SIG/Scheduling, we started to try to align how we compare things in tests with `cmp.Diff`, rather than having various ways to do so.
https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

Reply:
I've addressed your comments. Thanks for reviewing.